pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -367,6 +371,11 @@ public void sendAsync(Message<T> message, SendCallback callback) {
                     msgMetadataBuilder.setSequenceId(sequenceId);
                 } else {
                     sequenceId = msgMetadataBuilder.getSequenceId();
+                    if (sequenceId <= lastSequenceIdPushed) {
+                        callback.sendComplete(new PulsarClientException
+                                .InvalidMessageException("Message is definitely a duplicate"));
```

Comment:
This is not correct on 2 levels: 
 1. If `sequenceId <= lastSequenceIdPushed` we don't know yet whether the message is already dup, because the previous attempt might still fail. This has to be disambiguated by the broker which has visibility at the storage level.
 2. As mentioned before, we cannot throw error when there's a duplicate, rather we need to return "ok" to the application.